File: backend/src/main/java/com/accesodatos/service/BetService.java
package com.accesodatos.service;

import java.util.List;

import com.accesodatos.dto.bet.BetRequestDto;
import com.accesodatos.dto.bet.BetResponseDto;

public interface BetService {
	List<BetResponseDto> getAllBets();
	List<BetResponseDto> getBetsByUserId(Long userId);
	BetResponseDto createBet(BetRequestDto dto);
	BetResponseDto updateBetById(Long betId, BetRequestDto betRequestDto);
	void deleteBet(Long betId);	
	BetResponseDto getBetByUserIdAndByMatchId(Long userId, Long matchId);
	
}

File: frontend/src/service/bet.service.ts
import { BetRequest } from "../types/Bet";
import api from "./api";

const createBet = async (betRquest: BetRequest) => {
  const response = await api.post(`/bets`, betRquest);
  return response.data;
};

const getBetsByUserId = async (userId: number) => {
  const response = await api.get(`/bets/users/${userId}`);
  return response.data.data;
};

const deleteBet = async (betId: number) => {
  const response = await api.delete(`/bets/${betId}`);
  return response.data;
};

const getBetByUserAndMatch = async (userId: number, matchId: number) => {
  const response = await api.get(`/bets/users/${userId}/matches/${matchId}`);
  return response.data.data;
};

const updateBet = async (betId: number, betRequest: BetRequest) => {
  const response = await api.put(`/bets/${betId}`, betRequest);
  return response.data;
};

const BetService = {
  createBet,
  deleteBet,
  getBetByUserAndMatch,
  updateBet,
  getBetsByUserId,
};

export default BetService;
